plugins/image-snapshot/src/main.ts

```diff
@@ -0,0 +1,160 @@
+  matchImageSnapshot: (params?: ImageSnapshotOptions) => Promise<boolean>;
+};
+
+export interface ImageSnapshotPluginOptions {
+  /** Width of screenshot produced. */
+  imageWidth?: number;
+  /** Height of screenshot produced. */
+  imageHeight?: number;
+  /** Jest image snapshot parameters to be applied across all tests. */
+  globalMatchOptions?: MatchImageSnapshotOptions;
+  /* Function that returns a path which tells the plugin where to store snapshots */
+  getSnapshotsDir?: (parameters: { story: string, kind: string }) => string;
+};
+
+export default class ImageSnapshotPlugin implements ProofPlugin, CLIPlugin {
+  private updateSnapshots = false;
+  private getSnapshotsDir = getSnapshotsDir;
+  private imageWidth = 1280;
+  private imageHeight = 800;
+  private globalMatchOptions : MatchImageSnapshotOptions = {
+    customSnapshotIdentifier: createSnapshotIdentifier,
+    failureThresholdType: 'percent',
+    failureThreshold: 0.01,
+    blur: 1
+  };
+
+  constructor(options : ImageSnapshotPluginOptions = { globalMatchOptions: {} }) {
+    this.getSnapshotsDir = options.getSnapshotsDir ?? this.getSnapshotsDir;
+    this.imageWidth = options.imageWidth ?? this.imageWidth;
+    this.imageHeight = options.imageHeight ?? this.imageHeight;
+    this.globalMatchOptions = {
+      ...this.globalMatchOptions,
+      ...options.globalMatchOptions
+    };
+  }
+
+  apply(proof: Proof) {
+    const _globalMatchOptions = this.globalMatchOptions;
+    const _getSnapshotsDir = this.getSnapshotsDir;
+    const _imageWidth = this.imageWidth;
+    const _imageHeight = this.imageHeight;
+    const snapshotState = new SnapshotState('', {
+      updateSnapshot: this.updateSnapshots ? "all" : "new",
+      getPrettier: () => null,
+      getBabelTraverse: () => traverse
+    });;
+
+    proof.hooks.start.tap('image-snapshot', () => {
+      if (!fs.existsSync(SCREEN_SHOT_DIR)) {
+        fs.mkdirSync(SCREEN_SHOT_DIR);
+      }
+    });
+
+    proof.hooks.testStart.tap('image-snapshot', (t: ProofTest) => {
+      t.hooks.beforeExecute.tapPromise(
+        'image-snapshot',
+        async (_testFunc: any, testArgs: TestHookArgs) => {
+          testArgs.browser.addCommand(
+            'matchImageSnapshot',
+            async function(this: ImageSnapshotBrowser, {
+              windowWidth = 1280,
+              windowHeight = 800,
+              ...rest
+            } : ImageSnapshotOptions = {}) {
+              await this.setWindowSize(windowWidth, windowHeight);
+              
+              const testName = `${kebabCase(testArgs.config.kind)}--${kebabCase(testArgs.config.story)}`;
+              
+              const screenShotBuffer = await this.saveScreenshot(`${SCREEN_SHOT_DIR}/${testName}.png`);
+              
+              const resized = await sharp(screenShotBuffer)
+                .resize(_imageWidth, _imageHeight)
+                .png()
+                .toBuffer();
+
+              const snapshotDir = _getSnapshotsDir({
```

Comment:
If we don't include the filePath for the `ProofTest` instance we can add that in and use it to construct a `__image_snapshots__` directory next to the file being tested. That would hopefully alleviate the assumptions around the directory structure in the default `getSnapshotsDir` handler

Reply:
is this something for the future or is this possible now? @adierkens